### PR TITLE
Fix for #1822

### PR DESCRIPTION
**Pull request by egor-rogov**
### Problem statement

Sometimes you need to give the plain text a "console" look. Think of output of a program or results of a SQL query for an example. The text should look like code, but without any highlighting.

Highlightjs has several language names (defined by `noHighlightRe`: `no-highlight`, `nohighlight`, `plain`, `text`), one of which is described in `README.md`:

> To disable highlighting altogether use the `nohighlight` class:
> ```html
> <pre><code class="nohighlight">...</code></pre>
> ```

Unfortunately in this case `hljs` class is not added to `code`'s classes, hence the text doesn't look like other (highlighted) code.

### Solution proposed

The proposal is to change behavior of `plain` / `text` language: to make it act just like a normal language which highlights nothing. Any attempt of autodetection for this language will result in zero relevance for any input text. Hence `plain` / `text` language must be specified explicitly by the user, and autodetection tests must be skipped for it.

Specifically, this pull request adds:
1. New attribute `disableAutodetect`.
2. New language `plain` (alias `text`).
3. Changes for skipping autodetection and the corresponding test.
4. Removes `plain` and `text` from `noHighlightRe`; `no-?highlight` remains intact.

### CODE REVIEW

File: src/highlight.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ https://highlightjs.org/
       aliases   = {};
 
   // Regular expressions used throughout the highlight.js library.
-  var noHighlightRe    = /^(no-?highlight|plain|text)$/i,
+  var noHighlightRe    = /^no-?highlight$/i,
       languagePrefixRe = /\blang(?:uage)?-([\w-]+)\b/i,
```
Comment (marcoscaceres) [inline src/highlight.js]: Will removing plain and text affect existing users?

Comment (marcoscaceres) [reply src/highlight.js]: Hmm 🤔 We might need to search source files in other projects via Github search.  I suspect there will be some, even if not document. We should strive to remain backwards compatible.

Comment (egor-rogov) [reply src/highlight.js]: I'm skeptical about the idea of searching... Too easy to miss something.
Let's choose another name. How about `output` or `plaintext`?

Comment (marcoscaceres) [reply src/highlight.js]: `plaintext` works for me!

Comment (egor-rogov) [reply src/highlight.js]: Fine, I'll make changes.

Comment (marcoscaceres) [reply src/highlight.js]: Appropriate it. I’ll give others a chance to weigh in, but would like to merge by end of the week. Thanks for your hard work here and for thinking this through.

**Comment by marcoscaceres** [comment]
I’ll need to get the original maintainer to give me npm publish rights.

**Comment by qubyte** [comment]
Just dropping in to say thanks for this!

**Comment by RickStrahl** [comment]
So I'm back looking at this today and I still am not seeing the behavior I expected based on this discussion.

Adding the `plaintext` provider into my custom build the following works:

````
```plaintext
This is plain text
```
````

However, using something like this:

````
```text
This is plain text
```
````

or using `plain` or `''` still will render a plain tag without the `.hljs` style applied.

I thought I might be able to quick fix this by changing the `plaintext.js` like this and adding the aliases:

```js
function(hljs) {
    return {        
        aliases: ['text','txt','plain','plaintext'],
        disableAutodetect: true
    };
}
```

but that has no effect.

What I really need for my scenario is a `text`/`txt` handler that shows plain text, but does apply the `.hljs` class.

Am I missing something or is that still not possible even with this fix?

**Comment by egor-rogov** [comment]
Hi @RickStrahl,
it's still not possible and unlikely be possible in the future. For a quick-and-dirty fix you can find and edit this line in `highlight.js`:
```js
   var noHighlightRe    = /^(no-?highlight|plain|text)$/i,
```
Remove `|plain|text` from it, and together with your change to `plaintext.js` this should work.

**Comment by RickStrahl** [comment]
Thank you so much. The workaround works, but not really a long term solution given that this will change with every release unfortunately and I see no good way to automate that.

It would be nice to have some way to pass in the 'exclusion-list'.

For now I added a note to my custom build script with steps to manually change after updates which hopefully I'll remember.

Anyway - thank you for confirming. I remember when I followed the original discussion we had it wasn't clear that `text/plain` would not work and at least now I can either fix it

**Pull request merged**